### PR TITLE
PR #1 of backend API.

### PR DESCRIPTION
**Pull request by AbstractUmbra**
The basic structure of the API is done.

This includes all current paste endpoints:
- GET /paste/:id
- PUT /paste/:id
- PATCH /paste/:id
- DELETE /paste
- DELETE /paste/:id

The required auth backend was done by @IAmTomahawkx and tested too. The authentication implementation done to paste end points is documented and tested.
The small database implementation was spearheaded by @IAmTomahawkx and added on by myself. All has been tested so far and works as intended.
Model implementation returns expected data on queries.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+__pycache__/
+.vscode/
+mystbin/config.toml
```
Comment (EvieePy) [inline .gitignore:3]: Can the config be put in the root directory?

File: mystbin/rest/routers/admin.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+auth_model = HTTPBearer()
+
+
+@router.get("/admin/user/{user_id}", tags=["admin"], response_model=User, responses={
+    200: {"model": User},
+    401: {"model": Unauthorized}},
+    include_in_schema=False
+)
+async def get_any_user(request: Request, user_id: int, authorization: str = Depends(auth_model)) -> Union[JSONResponse, Dict[str, str]]:
+    """ Returns the User object of the passed user_id.
```
Comment (EvieePy) [inline mystbin/rest/routers/admin.py:38]: Not sure why there is a space at the start of docstrings

**Review by EvieePy — CHANGES_REQUESTED** [review]
That's all I have for now, there are some logic I would change but we can work on that later. Otherwise looks good 🙂

File: mystbin/rest/routers/admin.py (hunk 2 above)
Comment (AbstractUmbra) [reply mystbin/rest/routers/admin.py]: Resolved with 2478803.

File: .gitignore (hunk 1 above)
Comment (AbstractUmbra) [reply .gitignore]: Resolved with 11c2528.

**Comment by AbstractUmbra** [comment]
Closing this for now due to going back to the drawing board to improve cleanliness, database efficiency and further implementation details.

**Pull request closed without merging**